app/controllers/AddIncomeSourceController.scala

```diff
@@ -16,31 +16,72 @@
-import play.api.i18n.I18nSupport
-import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import controllers.predicates.{AuthenticationPredicate, IncomeSourceDetailsPredicate, NavBarPredicate, NinoPredicate, SessionTimeoutPredicate}
+import models.core.{AccountingPeriodModel, CessationModel}
+import models.incomeSourceDetails.{BusinessDetailsModel, IncomeSourceDetailsModel, PropertyDetailsModel}
+import play.api.mvc._
+import services.IncomeSourceDetailsService
+import uk.gov.hmrc.auth.core.AuthorisedFunctions
+import views.html.NewIncomeSources
+import models.incomeSourceDetails.viewmodels.IncomeSourcesViewModel
+import java.time.{LocalDate, Month}
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
 
-import javax.inject.Inject
-import scala.concurrent.ExecutionContext
+@Singleton
+class AddIncomeSourceController @Inject()(val newIncomeSources: NewIncomeSources,
+                                          val checkSessionTimeout: SessionTimeoutPredicate,
+                                          val authenticate: AuthenticationPredicate,
+                                          val authorisedFunctions: AuthorisedFunctions,
+                                          val retrieveNino: NinoPredicate,
+                                          val retrieveIncomeSources: IncomeSourceDetailsPredicate,
+                                          implicit val itvcErrorHandlerAgent: AgentItvcErrorHandler,
+                                          val incomeSourceDetailsService: IncomeSourceDetailsService,
+                                          val retrieveBtaNavBar: NavBarPredicate)
+                                         (implicit val ec: ExecutionContext,
+                                           implicit override val mcc: MessagesControllerComponents,
+                                           val appConfig: FrontendAppConfig) extends ClientConfirmedController
+  with FeatureSwitching {
 
-class AddIncomeSourceController @Inject()(val authenticate: AuthenticationPredicate,
-                                          val authorisedFunctions: FrontendAuthorisedFunctions,
-                                          val checkSessionTimeout: SessionTimeoutPredicate)
-                                         (implicit val appConfig: FrontendAppConfig,
-                                          mcc: MessagesControllerComponents,
-                                          val ec: ExecutionContext,
-                                          val itvcErrorHandlerAgent: AgentItvcErrorHandler)
-  extends ClientConfirmedController with FeatureSwitching with I18nSupport {
+  def show(): Action[AnyContent] = (checkSessionTimeout andThen authenticate andThen retrieveNino
+    andThen retrieveIncomeSources andThen retrieveBtaNavBar).async {
+    implicit user =>
+      handleRequest(
+        sources = user.incomeSources,
+        isAgent = false,
```

Comment:
really an optional, but I would keen to read isAgent state from user.userType, we can have some extension method on this type to distinguish this, but we might have to create a story to address rest of the page as well